### PR TITLE
add loggign to find NB_Global insert bug 

### PR DESCRIPTION
**Pull request by OmerBenHayun**

### CODE REVIEW

**Comment by OmerBenHayun** [comment]
example for logs from the north db pod that describe the bug:
image

File: pkg/ovsdb/transact.go [hunk 1]
```diff
@@ -1043,9 +1044,14 @@ func doInsert(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.O
 		}
 	}
 
+	klog.V(6).Infof("validate schema of DB %s on table %s", txn.request.DBName, ovsOp.Table)
 	ok, err := txn.schemas.Validate(txn.request.DBName, ovsOp.Table, &ovsOp.Row)
-	if !ok || err != nil {
-		klog.Errorf("Failed to validate table %s/%s", txn.request.DBName, ovsOp.Table)
+	if err != nil {
+		klog.Errorf("table %s validation returned %v", ovsOp.Table, err.Error())
```
Comment (hunchback) [inline pkg/ovsdb/transact.go:1050]: the error should be rather be:
```
klog.Errorf("Failed to validate table %s/%s: %s", txn.request.DBName, ovsOp.Table, err.Error())
```

as it now:
1. says that the validation failed (for example I could just be missing the schema while value is actually valid)
2. you are missing the database context <dbname>/<table>

File: pkg/ovsdb/transact.go [hunk 2]
```diff
@@ -1043,9 +1044,14 @@ func doInsert(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.O
 		}
 	}
 
+	klog.V(6).Infof("validate schema of DB %s on table %s", txn.request.DBName, ovsOp.Table)
```
Comment (hunchback) [inline pkg/ovsdb/transact.go:1047]: redundant - please remove

File: pkg/ovsdb/transact.go [hunk 3]
```diff
@@ -1043,9 +1044,14 @@ func doInsert(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.O
...
 	ok, err := txn.schemas.Validate(txn.request.DBName, ovsOp.Table, &ovsOp.Row)
-	if !ok || err != nil {
-		klog.Errorf("Failed to validate table %s/%s", txn.request.DBName, ovsOp.Table)
+	if err != nil {
+		klog.Errorf("table %s validation returned %v", ovsOp.Table, err.Error())
+		return errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	if !ok {
+		klog.Errorf("failed to validate table %s/%s", txn.request.DBName, ovsOp.Table)
```
Comment (hunchback) [inline pkg/ovsdb/transact.go:1054]: wrong message:

```
klog.Errorf("Row in table %s/%s is invalid", txn.request.DBName, ovsOp.Table)
```

File: pkg/libovsdb/schema.go [hunk 4]
```diff
@@ -736,6 +738,7 @@ func (columnSchema *ColumnSchema) ValidateMap(value interface{}) bool {
 }
 
 func (columnSchema *ColumnSchema) Validate(value interface{}) bool {
+	klog.V(5).Infof("validate columnSchema : %v value: %v", columnSchema, value)
```
Comment (hunchback) [inline pkg/libovsdb/schema.go:741]: remove: this is library code, it should not log message, this is for the caller program to decide

File: pkg/libovsdb/schema.go [hunk 5]
```diff
@@ -7,6 +7,8 @@ import (
 	"io/ioutil"
 	"os"
 	"strings"
+
+	"k8s.io/klog"
```
Comment (hunchback) [inline pkg/libovsdb/schema.go:11]: remove, library code should not log messages (this is for the caller program to decide)

File: pkg/libovsdb/schema.go [hunk 6]
```diff
@@ -766,7 +769,10 @@ func (tableSchema *TableSchema) Validate(row *map[string]interface{}) (bool, err
 		if column == "_uuid" || column == "_version" {
 			continue
 		}
+		klog.Infof("Validate row with column %s and value %s", column, value)
+		klog.Infof("we got the following table schema:%v", tableSchema)
 		columnSchema, ok := tableSchema.Columns[column]
+		klog.Infof("got the following column schema: %v", columnSchema)
```
Comment (hunchback) [inline pkg/libovsdb/schema.go:775]: remove logs

**Pull request merged**